File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
def solution(n, m, section):
    answer = 0
    
    paint = [True] * n
    
    for i in section:
        paint[i - 1] = False

    j = 0
    while j < n:
        if not paint[j]:
            answer += 1
            j += m
        else:
            j += 1
            
    return answer

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int[] wall = new int[n];
        
        for (int i = 0; i < section.length; i++) {
            wall[section[i] - 1] = 1;
        }
        
        int idx = 0;
        while (idx < n) {
            if (wall[idx] == 1) {
                answer++;
                idx += m;
            }
            else {
                idx++;
            }
        }
        
        return answer;
    }
}